cmake/GtsamBuildTypes.cmake

```diff
@@ -161,3 +146,23 @@ function(gtsam_assign_all_source_folders)
   gtsam_assign_source_folders("${all_c_srcs};${all_cpp_srcs};${all_headers}")
 endfunction()
 
+# Applies the per-config build flags to the given target.
+function(gtsam_apply_build_flags target_name_)
+  target_compile_definitions(${target_name_} PRIVATE ${GTSAM_COMPILE_DEFINITIONS_PRIVATE})
+  target_compile_definitions(${target_name_} PUBLIC ${GTSAM_COMPILE_DEFINITIONS_PUBLIC})
+  if (NOT "${GTSAM_COMPILE_OPTIONS_PUBLIC}" STREQUAL "")
+    target_compile_options(${target_name_} PUBLIC ${GTSAM_COMPILE_OPTIONS_PUBLIC})
+  endif()
+  target_compile_options(${target_name_} PRIVATE ${GTSAM_COMPILE_OPTIONS_PRIVATE})
+
+  # Also, ensure the compiler uses at least C++11.
+  # The use of target_compile_features() is preferred since it will be not in
+  # conflict with a more modern C++ standard, if used in a client program.
+  if(NOT CMAKE_VERSION VERSION_LESS 3.8)
```

Comment:
Can we replace double negative with one positive :-) ?

Reply:
Sure, I would prefer that too! :-)
The fact is that CMake ["if"](https://cmake.org/cmake/help/v3.5/command/if.html) doesn't provide a  greater-or-equal operator... There is the option of the double negative... or the trick of using `VERSION_GREATER 3.7.9` with the fictional version number (sigh). 

Preference here?

Reply:
just keep as is...